### ISSUE TITLE
PWCsurvOS returns values larger than 1

### ISSUE THREAD

**Issue by holgstr** [insightsengineering/simIDM#74]
**Summary**

For some configurations of arguments, PWCsurvOS may produce values slightly larger than 1.

```r

PWCsurvOS(3.13, c(0, 1, 1), c(0, 0.5, 1), c(0, 1, 1), c(0, 3, 8), c(0, 6, 7), c(0, 8, 9))

```

**R session info**

```sh

R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 11 x64 (build 22621)

Matrix products: default


locale:
[1] LC_COLLATE=German_Switzerland.utf8 
[2] LC_CTYPE=German_Switzerland.utf8   
[3] LC_MONETARY=German_Switzerland.utf8
[4] LC_NUMERIC=C                       
[5] LC_TIME=German_Switzerland.utf8    

time zone: Europe/Zurich
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets 
[6] methods   base     

other attached packages:
[1] simIDM_0.0.5.9011 testthat_3.1.10  
[3] devtools_2.4.5    usethis_2.2.2    

loaded via a namespace (and not attached):
 [1] stringi_1.7.12    lattice_0.21-8   
 [3] digest_0.6.33     magrittr_2.0.3   
 [5] evaluate_0.21     grid_4.3.1       
 [7] pkgload_1.3.2.1   fastmap_1.1.1    
 [9] rprojroot_2.0.3   Matrix_1.5-4.1   
[11] processx_3.8.2    pkgbuild_1.4.2   
[13] sessioninfo_1.2.2 brio_1.1.3       
[15] backports_1.4.1   survival_3.5-7   
[17] urlchecker_1.0.1  ps_1.7.5         
[19] promises_1.2.1    purrr_1.0.2      
[21] cli_3.6.1         shiny_1.7.5      
[23] rlang_1.1.1       crayon_1.5.2     
[25] ellipsis_0.3.2    splines_4.3.1    
[27] remotes_2.4.2.1   withr_2.5.0      
[29] cachem_1.0.8      yaml_2.3.7       
[31] tools_4.3.1       memoise_2.0.1    
[33] checkmate_2.2.0   httpuv_1.6.11    
[35] vctrs_0.6.3       R6_2.5.1         
[37] mime_0.12         lifecycle_1.0.3  
[39] stringr_1.5.0     fs_1.6.3         
[41] htmlwidgets_1.6.2 miniUI_0.1.1.1   
[43] desc_1.4.2        callr_3.7.3      
[45] later_1.3.1       glue_1.6.2       
[47] profvis_0.3.8     Rcpp_1.0.11      
[49] xfun_0.40         rstudioapi_0.15.0
[51] knitr_1.44        xtable_1.8-4     
[53] htmltools_0.5.6   rmarkdown_2.24   
[55] compiler_4.3.1    prettyunits_1.1.1

```

**OS / Environment**

- OS: Windows 10

**Comment by danielinteractive** [comment]
Thanks @holgstr , was this fixed by #76 already or not yet?

**Comment by holgstr** [comment]
@danielinteractive This has not been fixed yet.

**Comment by danielinteractive** [comment]
This was fixed now by #86 ?

**Comment by holgstr** [comment]
No, not sure where this comes from. There seem to be small numerical deviations:
```
library(ggplot2)

b1 <- 3
b2 <- 6
t_values <- seq(2.9, 6.2, length.out = 1000)
df <- data.frame(t = t_values, 
                 survival = PWCsurvOS(t_values, c(0, 1, 1), c(0, 0.5, 1), c(0, 1, 1), c(0, b1, 8), c(0, b2, 7), c(0, 8, 9)))

ggplot(df, aes(x = t, y = survival)) +
  geom_line() +
  labs(title = "PWC Survival Function", x = "t", y = "Survival Probability")          

```
Note the jumps in the proximity of `b1` and `b2`.

![PWCSurvOS](https://github.com/insightsengineering/simIDM/assets/64039523/0614b8fa-acbb-413d-bb61-7a8e1a2f1349)
